var/spack/repos/builtin/packages/py-notebook/package.py

```diff
@@ -24,16 +25,27 @@ class PyNotebook(PythonPackage):
     variant('terminal', default=False, description="Enable terminal functionality")
 
     depends_on('python@2.7:2.8,3.3:')
```

Comment:
The latest version only supports Python 3.5+. Also, need `type=('build', 'run')`.